gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -152,8 +153,27 @@ public static void movePath(FileSystem srcFs, Path src, FileSystem dstFs, Path d
     if (srcFs.getUri().equals(dstFs.getUri())) {
       renamePath(srcFs, src, dst);
     } else {
-      if (!FileUtil.copy(srcFs, src, dstFs, dst, true, false, dstFs.getConf())) {
-        throw new IOException(String.format("Failed to move %s to %s", src, dst));
+      boolean isSourceFileSystemLocal = srcFs instanceof LocalFileSystem;
```

Comment:
The benefit of determining that the source filesystem is local is that we can optimize how we move the files to the destination filesystem.  For instance, when copying to s3a filesystem the file is read via an inputstream into a buffer file at fs.s3a.buffer.dir.  Once the data has been copied to the buffer file, it is then uploaded to s3.  If we know that the source filesystem is local we can enable an optimization which skips copying to the buffer file.  This happens because s3a implements moveFromLocalFile to not copy to the buffer file first.

I'm not certain when a RawLocalFileSystem is used and therefore didn't check for that.  Additionally, the moveFromLocalFile expects that the file is in a LocalFileSystem.